app/controllers/annotations_controller.rb

```diff
@@ -0,0 +1,160 @@
+class AnnotationsController < ApplicationController
+
+  before_action :update_comments, only: [:edit, :update_annotations]
+
+  authorize_resource
+  
+  def new
+    @annotation = Annotation.new(category: :note, color: Annotation.colors[1])
+    @total_seconds = params[:total_seconds]
+    @medium_id = params[:mediumId]
+    @posted = (@annotation.public_comment_id != nil)
+    
+    render :edit
+  end
+
+  def edit
+    @annotation = Annotation.find(params[:annotationId])
+    # only allow editing, if the current user created the annotation
+    if @annotation.user_id != current_user.id
+      render json: false
+      return
+    end
+    @total_seconds = @annotation.timestamp.total_seconds
+    @medium_id = @annotation.medium_id
+    @posted = (@annotation.public_comment_id != nil)
+  end
+
+  def update
+    @annotation = Annotation.find(params[:id])
+    comment_id = @annotation.public_comment_id
+    if (comment_id == nil)
```

Comment:
Please make sure to use **rubocop** for stylistic hints (I get many more suggestions for this file for example). For this specific line, rubocop suggests to use `if comment_id.nil?`. Also: no parentheses around `if` statements in Ruby.

![image](https://github.com/MaMpf-HD/mampf/assets/37160523/8b52f47d-cff0-410d-88c9-ea3ee7a3987f)